pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -304,12 +309,19 @@ func NewTargetProxy(tcpBindAddress string, tcpBindPort int, libvirtdSocketPath s
 		stopChan:       make(chan struct{}),
 		fdChan:         make(chan net.Conn, 1),
 		listenErrChan:  make(chan error, 1),
+		tlsConfig:      tlsConfig,
 	}
 
 }
 
 func (m *migrationProxy) createTcpListener() error {
-	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", m.tcpBindAddress, m.tcpBindPort))
+	var listener net.Listener
+	var err error
+	if m.tlsConfig != nil {
+		listener, err = tls.Listen("tcp", fmt.Sprintf("%s:%d", m.tcpBindAddress, m.tcpBindPort), m.tlsConfig)
+	} else {
```

Comment:
The only reason TLS isn't used for all tcp connections is there's a special case were we have to bind to "127.0.0.1" to work around libvirt not providing us a way to supply a unix socket in a connection URI during migrations... hopefully that work around will go away some day. 

As an added precaution here, I think it makes sense to structure the logic like this.

```
if m.tlsConfig != nil {

} else if m.tcpBindAddress == "127.0.0.1" {

} else {
    return fmt.Errorf("TLS is required to bind to %s", m.tcpBindAddress)
}